CC/src/DgmOctree.cpp

```diff
@@ -967,8 +967,7 @@ void DgmOctree::getPointsInNeighbourCellsAround(NearestNeighboursSearchStruct &n
 						{
 							if (!getOnlyPointsWithValidScalar || ScalarField::ValidValue(m_theAssociatedCloud->getPointScalarValue(p->theIndex)))
 							{
-								PointDescriptor newPoint(m_theAssociatedCloud->getPointPersistentPtr(p->theIndex),p->theIndex);
-								nNSS.pointsInNeighbourhood.push_back(newPoint);
+								nNSS.pointsInNeighbourhood.emplace_back(PointDescriptor (m_theAssociatedCloud->getPointPersistentPtr(p->theIndex),p->theIndex));
```

Comment:
It's unclear to me whether doing it the way you have calls the copy constructor or not. It seems like it should because the prototype is:
```cpp
template< class... Args >
void emplace_back( Args&&... args );
```
And [the docs](http://en.cppreference.com/w/cpp/container/vector/emplace_back) say:

> The arguments args... are forwarded to the constructor as std::forward<Args>(args)....

All the docs and examples I've found use variadic args like a constructor (which I prefer in this case anyways).

Reply:
yep I made a test, you are totally right